contract/src/msgs/owner/execute/remove_from_allowlist.rs

```diff
@@ -9,10 +10,24 @@ impl ExecuteHandler for execute::remove_from_allowlist::Execute {
             return Err(ContractError::NotOwner);
         }
 
-        // remove the address from the allowlist
+        // we need to remove the address from the allowlist
         let public_key = PublicKey::from_hex_str(&self.public_key)?;
         ALLOWLIST.remove(deps.storage, &public_key);
 
+        if let Some(staker) = STAKERS.may_get_staker(deps.storage, &public_key)? {
+            // we move their staked tokens to the pending withdrawal
+            // so that they can withdraw them and no longer be a staker
+            let staker = Staker {
+                memo:                      staker.memo,
+                tokens_staked:             Uint128::new(0),
+                tokens_pending_withdrawal: staker.tokens_staked,
```

Comment:
Won't this remove any rewards they might already have accumulated?

Reply:
You totally right I didn't think about that.